assets/scripts/dialogs/UpgradeDialog.jsx

```diff
@@ -1,56 +1,118 @@
-import React, { Component } from 'react'
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { FormattedMessage, useIntl } from 'react-intl'
 import StripeCheckout from 'react-stripe-checkout'
-// import { FormattedMessage } from 'react-intl'
-import Dialog from './Dialog'
+import { connect } from 'react-redux'
+import axios from 'axios'
 import { STRIPE_API_KEY } from '../app/config'
-// import './UpgradeDialog.scss'
-
-class UpgradeDialog extends Component {
-  scriptContainerEl = React.createRef()
-
-  onToken = (token, addresses) => {
-    // TODO: Send the token information and any other
-    // relevant information to your payment process
-    // server, wait for the response, and update the UI
-    // accordingly. How this is done is up to you. Using
-    // XHR, fetch, or a GraphQL mutation is typical.
-    console.log('onToken', token, addresses)
+import userRoles from '../../../app/data/user_roles.json'
+import Dialog from './Dialog'
+import './UpgradeDialog.scss'
+
+const DEFAULT_BODY = 'Thank you for using Streetmix! For only $5/month, the Enthusiast Plan lets users support Streetmix while also gaining access to new experimental features. Plus your avatar gets a neat badge!'
+
+const UpgradeDialog = ({ userId, roles }) => {
+  const intl = useIntl()
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState(null)
+  const [data, setData] = useState(null)
```

Comment:
Using `useState` for three separate variables seems weird, but it's recommended by the docs:
https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables